contents/data/databases.md

```diff
@@ -0,0 +1,115 @@
+
+There are many types of database models. Some examples are:
+
+- [**Relational model**](https://en.wikipedia.org/wiki/Relational_model) - the most widely used database model, usually modeled using a table format. The data structure is defined by a set of relations and domains (types) that dictates the constraints. Constraints are also established for the type of operations that can be done.
+- [**Document-oriented**](https://en.wikipedia.org/wiki/Document-oriented_database) - data is stored in documents, that encapsulates the data. This data is normally stored in a semi-structured manner.
+- [**Flat**](https://en.wikipedia.org/wiki/Flat_file_database) - data stored as files, without any structure
+
+[Click here](https://en.wikipedia.org/wiki/Database_model) to learn more about the different database models.
+
+### Why are there different models?
+
+There are advantages and disadvantages to utilising one of the many database models available. These different models seek to solve certain issues when programs deal with data.
+
+For instance, the relational model is the most structured out of the three that were shown. The presence of structure allows the database to enhance and improve certain common operations, for instance, searching for a specific data for an entry would be faster.
+
+However, having such a rigid structure would mean that there's a limitation on how and what kind of data can be stored in the database.
+
+For the purposes of introduction, we would mainly cover on aspects that are used in the relational and/or document-oriented database models.
+
+## Relation
+
+The most popular database model, relational model, makes use of relations. This model assumes that the data to be stored follow a certain 'pattern'. For instance, a database that stores products sold by a shop would contain data such as: name, description, price and current stock levels. A visualisation of the database and data can be seen below:
+
+| Name  | Description                  | Price | Current Stock |
+|:-----:|:----------------------------:|:-----:|:-------------:|
+| Fruit | A fruit.                     | 1.00  |      100      |
+| Bread | Sliced for your convenience. | 1.40  |       50      |
+| Water | Essential for life.          | 0.50  |     1000      |
+
+From this visualisation, we can define the different parts of the relation:
+
+- The whole table is known as a **relation**.
+- Each data row (i.e. excluding the header row) is known as a **tuple**, for instance: (Fruit, A fruit., 1.00, 100).
+- Each column is an **attribute** and each attribute has a **domain** or **data type**. The *Current Stock* column of the table is an attribute with a data type of integer (as product stock is logically represented using an integer).
+- Each element in a tuple is called an **attribute value**
+
+[Click here](https://en.wikipedia.org/wiki/Relation_(database)) for a more in-depth and formal definition of relations.
```

Comment:
I'm no expert but I thought how data in multiple tables can relate to each other is also an important part of relational databases.